src/app.cpp

```diff
@@ -831,6 +831,21 @@ void configure(pcapng_exporter::PcapngExporter* exporter, AppText* obj) {
 	exporter->mappings.push_back(mapping);
 }
 
+template<class LinErrorBase>
+int write_lin_error(
+	pcapng_exporter::PcapngExporter writer,
+	LinErrorBase *lerr,
+	std::uint8_t errors,
+	uint64_t date_offset_ns)
+{
+	pcapng_exporter::frame_header header = generate_header(lerr, date_offset_ns);
+	if (header.timestamp_resolution == 0) return -3;
```

Comment:
What's this for?

Reply:
This generates the header that is necessary for lin message and error writing in pcapng.
Since the program is working with error return codes the time resolution error code is returned in the write function(s). The appropriate error message is printed when it arises.